src/Voron/Impl/Journal/JournalFile.cs

```diff
@@ -6,62 +6,82 @@
     {
         public long LastTransactionId;
 
-        internal List<TransactionHeader> _transactionHeaders = new();
+        public readonly Dictionary<StorageEnvironment, JournalFile> RegisteredEnvironments = new();
 
+        private List<TransactionHeader> _transactionHeaders = new();
+
+        public FrozenSet<Guid> RecoveredJournalIds = recoveredJournalIds;
+        
         public override string ToString()
         {
             return $"Number: {Number}";
         }
 
-        internal long GetWritePosIn4KbPosition(EnvironmentStateRecord record) => record.Journal.Current == this ? record.Journal.Last4KWritePosition : 0;
+        internal long GetWritePosIn4KbPosition(EnvironmentStateRecord record) => record.Journal.Number == Number ? record.Journal.Last4KWritePosition : 0;
 
         public long Number { get; } = journalNumber;
 
+        public SingleUseFlag DoneWriting;
+        private JournalWriter _journalWriter = journalWriter;
+        public bool NewlyCreatedFile;
 
-        public long GetAvailable4Kbs(EnvironmentStateRecord record) => (journalWriter?.NumberOfAllocated4Kb - GetWritePosIn4KbPosition(record)) ?? 0;
+        public long GetAvailable4Kbs(EnvironmentStateRecord record) => (_journalWriter?.NumberOfAllocated4Kb - GetWritePosIn4KbPosition(record)) ?? 0;
 
-        public Size JournalSize => new Size(journalWriter?.NumberOfAllocated4Kb * 4 ?? 0, SizeUnit.Kilobytes);
+        public Size JournalSize => new Size(_journalWriter?.NumberOfAllocated4Kb * 4 ?? 0, SizeUnit.Kilobytes);
 
-        internal JournalWriter JournalWriter => journalWriter;
+        internal JournalWriter JournalWriter => _journalWriter;
 
         public void Release()
         {
-            if (journalWriter?.Release() != true)
+            if (_journalWriter?.Release() != true)
                 return;
 
             Dispose();
         }
 
         public void AddRef()
         {
-            journalWriter?.AddRef();
+            _journalWriter?.AddRef();
         }
 
         public void Dispose()
         {
             _transactionHeaders = null;
-            journalWriter = null;
+            _journalWriter = null;
         }
 
-        public void SetLastReadTxHeader(long maxTransactionId, ref TransactionHeader lastReadTxHeader)
+        public (long FirstTransactionId, long LastTransactionId, int Count)  GetTransactionStatsFor(Guid journalId)
+        {
+            int count = 0;
+            long first = long.MaxValue;
+            long last = -1;
+            for (int i = 0; i < _transactionHeaders.Count; i++)
+            {
+                if(_transactionHeaders[i].JournalId != journalId)
+                    continue;
+                count++;
+                first = Math.Min(_transactionHeaders[i].TransactionId, first);
+                last = _transactionHeaders[i].TransactionId;
```

Comment:
We're looking for the lowest tx id because they might come from different environments, right? So we have `first = Math.Min()`. Why we don't do `Math.Max()` for `last`?